### ISSUE TITLE
[BUG] Unhandled exception: type 'Null' is not a subtype of type 'Object' in type cast

### ISSUE THREAD

**Issue by zs-dima** [dart-code-checker/dart-code-metrics#422]
Reproduced with dart_code_metrics versions:
  dart_code_metrics: ^4.1.0
  dart_code_metrics: ^4.2.0-dev.1
  dart_code_metrics: ^4.2.0-dev.2

dart_code_metrics setup according documentation

> flutter --version
```
Flutter 2.5.0-5.0.pre • channel dev • https://github.com/flutter/flutter.git
Framework • revision 0f465e5b2a (9 days ago) • 2021-08-04 18:59:50 -0400
Engine • revision a0d89b1a54
Tools • Dart 2.14.0 (build 2.14.0-377.0.dev)
```

exception appear on:
> flutter pub run dart_code_metrics:metrics lib
or 
> dart pub run dart_code_metrics:metrics lib

**What actually happened?**
> flutter pub run dart_code_metrics:metrics lib
```
Unhandled exception:
type 'Null' is not a subtype of type 'Object' in type cast
#0      yamlScalarToDartObject (package:dart_code_metrics/src/utils/yaml_utils.dart:33:66)
#1      yamlNodeToDartObject (package:dart_code_metrics/src/utils/yaml_utils.dart:26:14)
#2      yamlMapToDartMap.<anonymous closure> (package:dart_code_metrics/src/utils/yaml_utils.dart:14:15)
#3      MappedIterator.moveNext (dart:_internal/iterable.dart:391:20)
#4      MapMixin.addEntries (dart:collection/maps.dart:177:23)
#5      new Map.fromEntries (dart:core/map.dart:187:17)
#6      yamlMapToDartMap (package:dart_code_metrics/src/utils/yaml_utils.dart:9:22)
#7      yamlNodeToDartObject (package:dart_code_metrics/src/utils/yaml_utils.dart:22:14)
#8      yamlMapToDartMap.<anonymous closure> (package:dart_code_metrics/src/utils/yaml_utils.dart:14:15)
#9      MappedIterator.moveNext (dart:_internal/iterable.dart:391:20)
#10     MapMixin.addEntries (dart:collection/maps.dart:177:23)
#11     new Map.fromEntries (dart:core/map.dart:187:17)
#12     yamlMapToDartMap (package:dart_code_metrics/src/utils/yaml_utils.dart:9:22)
#13     _loadConfigFromYamlFile (package:dart_code_metrics/src/config_builder/models/analysis_options.dart:120:27)
#14     analysisOptionsFromFile (package:dart_code_metrics/src/config_builder/models/analysis_options.dart:110:33)
#15     analysisOptionsFromContext (package:dart_code_metrics/src/config_builder/models/analysis_options.dart:99:9)
#16     LintAnalyzer.runCliAnalysis (package:dart_code_metrics/src/analyzers/lint_analyzer/lint_analyzer.dart:85:37)
#17     AnalyzeCommand.runCommand (package:dart_code_metrics/src/cli/commands/analyze.dart:59:48)
#18     BaseCommand._verifyThenRunCommand (package:dart_code_metrics/src/cli/commands/base_command.dart:93:12)
#19     BaseCommand.run (package:dart_code_metrics/src/cli/commands/base_command.dart:25:25)
#20     CommandRunner.runCommand (package:args/command_runner.dart:209:27)
#21     CommandRunner.run.<anonymous closure> (package:args/command_runner.dart:119:25)
#22     new Future.sync (dart:async/future.dart:218:31)
#23     CommandRunner.run (package:args/command_runner.dart:119:14)
#24     CliRunner.run (package:dart_code_metrics/src/cli/cli_runner.dart:28:19)
#25     main (file:///C:/Users/a/AppData/Local/Pub/Cache/hosted/pub.dartlang.org/dart_code_metrics-4.2.0-dev.2/bin/metrics.dart:4:21)
#26     _delayEntrypointInvocation.<anonymous closure> (dart:isolate-patch/isolate_patch.dart:281:32)
#27     _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:184:12)
pub finished with exit code 255

```

**Comment by dkrutskikh** [comment]
@zs-dima please attach your analysis_options.yaml

**Comment by zs-dima** [comment]
Looks like analysis_options.yaml issue, sorry, works fine after flutter update and analysis_options.yaml re-create from documentation

**Comment by johnnyasantoss** [comment]
Having an empty `linter` prop on root of the yaml file makes this error appear.
Like so:

```yaml
include: package:flutter_lints/flutter.yaml

linter:
  # nothing really, just empty like the one on flutter docs

# ... dart_code_metrics default configs from the README
```

**Comment by dkrutskikh** [comment]
@johnnyasantoss please attach your analysis_options.yaml

**Comment by johnnyasantoss** [comment]
```yaml
include: package:flutter_lints/flutter.yaml

linter:
  # rules:
    # avoid_print: true

analyzer:
   plugins:
     - dart_code_metrics

dart_code_metrics:
  anti-patterns:
    - long-method
    - long-parameter-list
  metrics:
    cyclomatic-complexity: 20
    maximum-nesting-level: 5
    number-of-parameters: 4
    source-lines-of-code: 50
  metrics-exclude:
    - test/**
    - lib/generated/**
  rules:
    - newline-before-return
    - no-boolean-literal-compare
    - no-empty-block
    - prefer-trailing-comma
    - prefer-conditional-expressions
    - no-equal-then-else

```

**Comment by dkrutskikh** [comment]
@johnnyasantoss thanks - I can repro, I'm taking a look.

**Comment by dkrutskikh** [comment]
@johnnyasantoss fixed in version 4.3.3. Could you please check?

**Comment by johnnyasantoss** [comment]
@dkrutskikh it works perfectly now :)
